app/services/authService.js

```diff
@@ -115,6 +117,42 @@ class AuthService {
     };
   }
 
+  async sendActivateEmail(userId, userIp) {
+    const oUser = await PortalUserModel.findById(userId).lean();
+    if (!oUser) {
+      throw new CustomError(STATUS_TYPE.PORTAL_USER_NOT_FOUND);
+    }
+    if (oUser.is_activated) {
+      throw new CustomError(STATUS_TYPE.PORTAL_USER_ALREADY_ACTIVATED);
+    }
+    // generate Token
+    const code = await this._getCode(oUser, userIp);
+    const siteName = config.siteName;
+    const siteUrl = config.siteUrl;
+    const activateUrl = `${siteUrl}/active-confirm/${code}`;
+    const displayName = oUser.nick_name || oUser.email;
+    const welcomMailPath = path.resolve(__dirname,'../views/welcome_mail.html');
+
+    const html = await ejs.renderFile( welcomMailPath, { siteName, displayName, activateUrl, siteUrl });
+    const activeEmail = {
+      async: false,
+      to: [oUser.email],
+      subject: `Welcome to Register at ${siteName}`,
+      text: `Hello ${displayName}, welcome to register. Please click the following link to activate your account: ${activateUrl}`,
+      html,
+    };
+
+    const sendEmailRes = await EmailService.sendEmail(activeEmail);
+    if (sendEmailRes.emailStatus == 0) {
+      // TODO error log
+      console.error('Error sending activation email:', sendEmailRes.desc);
+    }
+
+    return {
+      'message': 'Activation email will be sent!'
+    };
+  }
+
   async _verifyPassword(password, salt, encryptedPwd) {
     const iteration = 1000;
```

Comment:
Why there is a magic number 1000? What's that for?

Reply:
Because we hash the password for 1000 times. So in this function we also need to set the iteration is 1000.